v5e_high_performance.md

```diff
@@ -0,0 +1,139 @@
+     to quickly get code up and running for fast experimentation or 
+     [multihost_job.py](https://github.com/google/maxtext/blob/main/README.md#getting-started-production-jobs-on-multiple-slices) for longer training runs. If you use these tools, 
+     then use the above as the input to `--COMMAND`, e.g.:
+     ```
+     python3 multihost_runner.py --TPU_PREFIX=${TPU_PREFIX} --COMMAND="bash setup.sh && bash MaxText/configs/largest_job/128b.sh RUN_NAME=${YOUR_RUN_NAME} OUTPUT_PATH=${MAXTEXT_OUTPUT_PATH} DATASET_PATH=${MAXTEXT_DATASET_PATH} PLATFORM=gce"
+     ```
+
+### Running on GKE with XPK
+These instructions show how to run the model configurations using XPK on top of GKE. More information and usage of XPK can be found [here](https://github.com/google/maxtext/tree/main/xpk).
+1. Create a custom MTU network to optimize network performance and give it firewall rules. If you are unable to complete this step, you may skip it. This step is not necessary, however performance degradation from scaling up the number of v5e pods over DCN will become more noticable.
+     
+     Create a network with an MTU of 8896 bytes and set up firewall rules. (Creating a network requires `compute.networks.create` permission in your project)
+     ```
+     gcloud compute networks create mtu9k --mtu=8896 --project={PROJECT} --subnet-mode=auto --bgp-routing-mode=regional
+     ```
+     ```
+     gcloud compute firewall-rules create mtu9kfw --network mtu9k --allow tcp,icmp,udp --project={PROJECT}
+     ```
+
+2. Install MaxText on your runner.
+     ```
+     # Install maxtext
+     git clone git@github.com:google/maxtext.git
+     ```
+
+
+3. Download dataset and set up GCS paths. If you have not downloaded the dataset before, we recommend following [README.md](https://github.com/google/maxtext/blob/main/README.md#getting-started-download-dataset-and-configure).
+
+4. Create a GKE cluster with the custom network created in Step 1.
+     ```
+     python3 xpk/xpk.py cluster create --cluster ${YOUR_CLUSTER_NAME} --num-slices=${NUM_SLICES} --tpu-type=${ACCELERATOR_TYPE} --zone=${ZONE} --project=${PROJECT} --custom-cluster-arguments="--network=mtu9k --subnetwork=mtu9k"
+     ```
+
+     If you did not create a custom network in Step 1, create a GKE cluster by running the same command without the `--network` and `--subnetwork` flags.
+     ```
+     python3 xpk/xpk.py cluster create --cluster ${YOUR_CLUSTER_NAME} --num-slices=${NUM_SLICES} --tpu-type=${ACCELERATOR_TYPE} --zone=${ZONE} --project=${PROJECT}
+     ```
+
+5. On your runner, build a MaxText dependency docker image.
+     ```
+     bash docker_build_dependency_image.sh
+     ```
+
+6. Upload the docker image to your GCP project.
+     ```
+     # Be sure to have your project and compute/zone set
+     gcloud config set project ${PROJECT}
+     gcloud config set compute/zone ${ZONE}
+
+     # Upload docker image
+     bash docker_upload_runner.sh CLOUD_IMAGE_NAME=${YOUR_IMAGE_NAME}
+     ```
+
```

Comment:
Not sure if you want to add a step  (6.5) about  `xpk cluster cacheimage` to enable faster start times.

Reply:
https://github.com/google/maxtext/blob/main/xpk/README.md#cluster-cacheimage
and https://github.com/google/maxtext/blob/main/xpk/xpk-large-scale-guide.sh#L482